File: extensions/panache/panache-jpa/runtime/src/main/java/io/quarkus/panache/jpa/PanacheRepository.java
package io.quarkus.panache.jpa;

/**
 * <p>
 * Represents a Repository for a specific type of entity {@code Entity}, with an ID type
 * of {@code Long}. Implementing this repository will gain you the exact same useful methods
 * that are on {@link PanacheEntityBase}. If you have a custom ID strategy, you should
 * implement {@link PanacheRepositoryBase} instead.
 * </p>
 *
 * @author Stéphane Épardaud
 * @param <Entity> The type of entity to operate on
 */
public interface PanacheRepository<Entity> extends PanacheRepositoryBase<Entity, Long> {

}

File: extensions/panache/hibernate-orm-panache-kotlin/runtime/src/main/kotlin/io/quarkus/hibernate/orm/panache/kotlin/PanacheRepository.kt
package io.quarkus.hibernate.orm.panache.kotlin

/**
 * Represents a Repository for a specific type of entity `Entity`, with an ID type of `Long`. Implementing this
 * interface will gain you the exact same useful methods that are on [PanacheEntity] and [PanacheCompanion]. If you
 * have a custom ID strategy, you should implement [PanacheRepositoryBase] instead.
 *
 * @param Entity The type of entity to operate on
 */
interface PanacheRepository<Entity : Any>: PanacheRepositoryBase<Entity, Long>